eng/pipelines/templates/pipeline.yml

```diff
@@ -1,6 +1,22 @@
 jobs:
+- job: Initialize_Build_SemanticVersion
+  timeoutInMinutes: 2
+  pool:
+    name: VSEng-MicroBuildVS2019
+    demands:
+      - DotNetFramework
+      - msbuild
+
+  steps:
+  - template: Initialize_Build_SemanticVersion.yml
+
 - job: Initialize_Build
+  dependsOn: Initialize_Build_SemanticVersion
   timeoutInMinutes: 10
+  variables:
+    SemanticVersion: $[dependencies.Initialize_Build_SemanticVersion.outputs['setsemanticversion.SemanticVersion']]
+    BuildRevision: $[counter(format('{0}.{1}', variables['SemanticVersion'], variables['build.definitionname']), 7130)]
```

Comment:
> I added a new job the CI pipeline which could increase CI duration by a minute due to the dependency on fetching semantic version from config.props using msbuild.exe.

This could be a LOT more than 1 minute if all the agents in the pool are busy.

Rather than making these job variables, can you define the variable in the step definition, and therefore move the "Set Semantic Version" to the same job as where it's used?

Reply:
I forgot to mention in the PR description, but I tried several ways to keep `Set Semantic Version` in `Initialize_Build` job instead of creating a new job. I couldn't figure out a best way because [doc](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/expressions?view=azure-devops#counter) says `counter function can only be used in an expression that defines a variable`.  This [doc](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch#user-defined-variables) says, In YAML pipelines, user-defined variables can set variables at the `root, stage, and job` level and in the UI. We execute a [PowerShell script in CI pipeline](https://github.com/NuGet/NuGet.Client/blob/dev/eng/pipelines/templates/Initialize_Build.yml#L43) to get the semantic version from [config.props](https://github.com/NuGet/NuGet.Client/blob/dev/build/config.props#L18). To satisfy the requirement for using `counter` expression, I ended up creating a new CI job and set a dependency for `Initialize_Build` job.

AFAIK, if we define `semantic version` as a variable in the `.yaml` file instead of reading from `config.props` which eliminates the need to execute a PowerShell script to fetch it, then we can remove `Intialize_Build_SemanticVersion` job completely.

Reply:
ok, makes sense. I thought/hoped it was possible to define variables at the task level, but it does make sense that a "job" is the lowest level it can be set.

I believe with your change that neither your new job, nor the existing "initialize build" job need the fileshare, so can you please try changing `name: VSEng-MicroBuildVS2019` to `vmImage: windows-latest` to see if it works on a "hosted" agent, so we're not affected when the microbuild pool is full.

Reply:
Changed `Initialize_Build_SemanticVersion` and `Initialize_Build` jobs to run on `vmImage: windows-latest`.  These two jobs have passed on the [CI with build number 5.10.0.7132](https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=4473701&view=results).